File: src/main/java/com/example/dai01/leetcode_977/Solution.java
package com.example.dai01.leetcode_977;

/**
 * 代码随想录day01第三题,有序数组的平方
 */
public class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int[] arr = new int[n];
        int l = 0,r = n - 1;
        while(l <= r){
            int ansl = nums[l] * nums[l];
            int ansr = nums[r] * nums[r];
            if(ansl >= ansr){
                arr[n - 1] = ansl;
                l ++;
            }else{
                arr[n - 1] = ansr;
                r --;
            }
            n--;//一定要记得减一下,要不就全覆盖到n - 1上了
        }
        return arr;
    }
}

File: src/main/java/com/example/dai01/leetcode_977/Solution.py
from typing import List


class Solution:
    def sortedSquares(self,nums : List[int]) -> List[int]:
        l,r = 0,len(nums) - 1
        ans = [0] * len(nums)
        while l <= r:
            if abs(nums[l]) > abs(nums[r]):
                ans[r - l] = nums[l] * nums[l]
                l += 1
            else:
                ans[r - l] = nums[r] * nums[r]
                r -= 1
        return ans